File: v3-sandbox/prototype-keys-java/src/main/java/org/hiero/keys/PrivateKey.java
package org.hiero.keys;

import org.hiero.keys.impl.KeyFactory;
import org.hiero.keys.io.ByteImportEncoding;
import org.hiero.keys.io.KeyFormat;

import static org.hiero.keys.io.KeyFormat.PKCS8_WITH_PEM;

public interface PrivateKey extends Key {
    byte[] sign(final byte[] message);

    PublicKey createPublicKey();

    static PrivateKey generate(final KeyAlgorithm algorithm) {
        return KeyFactory.generatePrivateKey(algorithm);
    }

    static PrivateKey create(final KeyAlgorithm algorithm, final byte[] rawBytes) {
        return KeyFactory.createPrivateKey(algorithm, rawBytes);
    }

    static PrivateKey create(final KeyAlgorithm algorithm, final ByteImportEncoding encoding, final String value) {
        return KeyFactory.createPrivateKey(algorithm, encoding, value);
    }

    static PrivateKey create(final KeyFormat format, final byte[] value) {
        return KeyFactory.createPrivateKey(format, value);
    }

    static PrivateKey create(final KeyFormat format, final String value) {
        return KeyFactory.createPrivateKey(format, value);
    }

    static PrivateKey create(final String value) {
        return create(PKCS8_WITH_PEM, value);
    }
}

File: v3-sandbox/prototype-python/src/hiero_sdk_v3/keys/private_key.py
from __future__ import annotations
from abc import abstractmethod
from .key import Key
from .key_algorithm import KeyAlgorithm
from .io.key_format import KeyFormat
from .public_key import PublicKey


class PrivateKey(Key):
    @abstractmethod
    def sign(self, message: bytes) -> bytes:
        ...

    @abstractmethod
    def create_public_key(self) -> PublicKey:
        ...

    @staticmethod
    def generate(algorithm: KeyAlgorithm) -> PrivateKey:
        from ._impl import generate_private_key
        return generate_private_key(algorithm)

    @staticmethod
    def create(algorithm: KeyAlgorithm, raw_bytes: bytes) -> PrivateKey:
        from ._impl import create_private_key_from_raw
        return create_private_key_from_raw(algorithm, raw_bytes)

    @staticmethod
    def create_from_format(format: KeyFormat, value: bytes | str) -> PrivateKey:
        from ._impl import create_private_key_from_format
        return create_private_key_from_format(format, value)

    @staticmethod
    def from_pem(pem: str) -> PrivateKey:
        return PrivateKey.create_from_format(KeyFormat.PKCS8_WITH_PEM, pem)
